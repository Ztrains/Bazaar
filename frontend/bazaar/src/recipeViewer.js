import React from 'react'
//import './index.css'
import RecipeEntry from './recipeEntry';
import axios from 'axios';

export default class RecipeViewer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      list: [],
    }
  }
  componentDidMount() {
    var _this = this;
    axios.get("https://bazaar-408.herokuapp.com/recipes")
    .then(function(result) {
      console.log(result);
      if (result === null) {
        alert("there is no data");

      }
      else {
      _this.setState({
        list: result.data,
      });
    }
    });
  }
  createList() {
    let newList = [];
    for (var i = 0; i < this.state.list.length; i++) {
      console.log(this.state.list[i]);
      newList.push(
        <div><RecipeEntry name={this.state.list[i].name} description={this.state.list[i].description} id={this.state.list[i].id}/></div>
      );
    }
    return newList;
  }
  render() {
    return(
      <div>


        <div className="card">
          {this.state.list.map((item, index) => {
            return <RecipeEntry name={item.name} />
          })}
        </div>
        </div>
    );
  }
}
