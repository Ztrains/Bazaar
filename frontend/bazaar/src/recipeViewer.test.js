import React from 'react';
import { shallow } from 'enzyme';
import recipeViewer from './recipeViewer.js';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('Testing if recipeViewer correctly renders', () => {
    const wrapper = shallow(<recipeViewer/>);
    const fetched = true;

    expect(toJson(wrapper)).toMatchSnapshot();
});