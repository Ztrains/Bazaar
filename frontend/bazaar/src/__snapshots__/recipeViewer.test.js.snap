// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if recipeViewer correctly renders 1`] = `<recipeViewer />`;
